client/components/marked-lines/index.js

```diff
@@ -0,0 +1,70 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classNames from 'classnames';
+import { map } from 'lodash';
+
+/**
+ * Translates marked-file context input
+ * into React component output
+ *
+ * @example
+ * const marks = [ [ 2, 4 ], [ 5, 9 ] ]
+ * const content = '->^^-_____<--'
+ * markup( marks, content ) === [ '->', <mark>{ '^^' }</mark>, '-', <mark>{ '_____' }</mark>, '<--' ]
+ *
+ * @param {Array<Array<number>>} marks spanning indices of text to mark, values in UCS-2 code units
+ * @param {string} content the plaintext content to mark
+ * @return {Array|string} list of output text nodes and mark elements or plain string output
+ */
+const markup = ( marks, content ) => {
+	if ( ! marks ) {
+		return content;
+	}
+
+	const mark = text => (
+		<mark key={ text } className="marked-lines__mark">
+			{ text }
+		</mark>
+	);
+
+	const [ finalOutput, finalLast ] = marks.reduce(
```

Comment:
This function is quite hard to grok. Maybe some comments would help?

Reply:
you mean other than the docblock comments? I should move `mark` out of here so it doesn't cloud this function…

Reply:
rearranged and added comments in 08244a217f - I hope that helps but please feel free to chime in more